### DISCUSSION TITLE
Extend user profile

### DISCUSSION THREAD

**Discussion by phpai** [openkfw/TruBudget#862] [General]
In GitLab by @jzakotnik on Feb 1, 2019, 10:13

User profile should contain
- name (already there)
- password (already there)
- email address
- organisation unit (optional)

**Comment by phpai** [comment]
In GitLab by @jzakotnik on Feb 1, 2019, 10:13

relevant for BNDES and Burkina

**Comment by phpai** [comment]
In GitLab by @kevinbader on Feb 18, 2019, 10:50

- [ ] new fields as described
- [ ] new intents for changing these fields

**Comment by phpai** [comment]
In GitLab by @kevinbader on Feb 20, 2019, 11:53

assigned to @mathiashoeld

**Comment by phpai** [comment]
In GitLab by @lennart.ploen on Mar 12, 2019, 14:00

Strong demand, so if this is easily implemented lets put it on high prio -> Quick win

**Comment by gonzochic** [comment]
There are countries which do not have a GDPR regulation. Therefore we should make the Email and the Organization Field available but **optional**

**Comment by kevinbader** [comment]
Perhaps it would make to store this off-chain, as the change history of things like email addresses or profile pictures doesn't need to be recorded. Idea outline:

- the API provides endpoints for querying and updating profile information
- an organization is only allowed to update information for its own users
- anyone can query the (public?) information of any user in the network (the API employs standard HTTP caching headers)

Within an organization, the profile information should be stored in a distributed way, perhaps using multi-node CockroachDB or something similar.

Pros:

- Organizations manage their own data; consequently, they can enforce any region-specific regulations (e.g. GDPR).
- Organizations cannot see previous profile data of foreign users. The history of this data is irrelevant for our use cases, so there's no need to share it.

Cons:

- Additional dev and ops complexity.
